servers/src/main/java/tachyon/worker/block/meta/StorageTier.java

```diff
@@ -147,4 +148,17 @@ public StorageDir getDir(int dirIndex) {
   public List<StorageDir> getStorageDirs() {
     return Collections.unmodifiableList(mDirs);
   }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj != null && obj instanceof StorageTier) {
+      return mTierLevel == ((StorageTier) obj).mTierLevel;
+    }
+    return false;
+  }
+
+  @Override
+  public int hashCode() {
+    return mTierLevel;
+  }
```

Comment:
Yes, this confused me at first because in actual case of creating the tiered block store, the default `equal` methods is just enough for possible tiers or dirs to check equals or not. While i still added it in this PR for better hearing your opinions.